staging/src/k8s.io/client-go/tools/portforward/portforward.go

```diff
@@ -37,7 +38,15 @@ import (
 // TODO move to API machinery and re-unify with kubelet/server/portfoward
 const PortForwardProtocolV1Name = "portforward.k8s.io"
 
-var ErrLostConnectionToPod = errors.New("lost connection to pod")
+var (
+	// error returned whenever we lost connection to a pod
+	ErrLostConnectionToPod = errors.New("lost connection to pod")
+
+	// set of error we're expecting during port-forwarding
+	networkClosedError = "use of closed network connection"
+	networkClosedRe    = regexp.MustCompile(`network namespace for sandbox "\w+" is closed`)
+	streamTimeoutRe    = regexp.MustCompile(`error creating forwarding stream for port \d+ -> \d+: Timeout occurred`)
```

Comment:
I assume these match specific error formatting in specific spots in the kubelet or container runtimes? can we:
1. link to where those error messages exist from here
2. verify those message strings match against all ~current versions of those runtimes
3. add comments / tests to those places in those code bases that these error string formats are now load-bearing and must not be changed?

I'd also consider making these slightly more lenient in how they match (like `network namespace for sandbox .* is closed` and `error creating forwarding stream for port`)

Reply:
ACK, will make sure to add these here.

Reply:
I've added them with links to particular piece of code for both containerd and cri-o.